File: cpp/number1.cpp
#include <iostream>

using namespace std;

void startPosFinder(int m, int n, const char* s){
  int x = 0;
  int y = 0;
  int rx = 0;
  int lx = 0;
  int uy = 0;
  int dy = 0;
  for(int i = 0; s[i] != '\0'; i++){
    if(s[i] == 'U'){
      y--;
      if (y < uy){
        uy = y;
      }
    }else if(s[i] == 'D'){
      y++;
      if (y > dy){
        dy = y;
      }
    }else if(s[i] == 'L'){
      x--;
      if (x < lx){
        lx = x;
      }
    }else if(s[i] == 'R'){
      x++;
      if (x > rx){
        rx = x;
      }
    }
  }
  if (rx - lx >= m || dy - uy >= n)
  {
    printf("No");
  }else
  {
    printf("%d %d", 1 - lx, 1 - uy);
  }
}

int main(){
  const char* s = "ULDDRRUU";
  startPosFinder(3, 3, s);
  return 0;
}

File: rb/number1.rb
m = 3
n = 3
s = "ULDDRRUU"

def start_pos_finder(m, n, s)
  x = 0
  y = 0
  r_x = 0
  l_x = 0
  u_y = 0
  d_y = 0

  s.each_char do |i|
    case i
    when "D"
      y += 1
      d_y = y if y > d_y
    when "U"
      y -= 1
      u_y = y if y < u_y
    when "L"
      x -= 1
      l_x = x if x < l_x
    when "R"
      x += 1
      r_x = x if x > r_x
    end
  end

  if r_x - l_x >= m || d_y - u_y >= n
    return "No"
  end

  [1 - l_x, 1 - u_y]
end

puts start_pos_finder(m, n, s)